lib/api/multiObjectDelete.js

```diff
@@ -233,6 +233,12 @@ function getObjMetadataAndDelete(authInfo, canonicalID, request,
                         log.debug('trying to delete locked object');
                         return callback(objectLockedError);
                     }
+                    if (versionId && objMD.location &&
+                        Array.isArray(objMD.location) && objMD.location[0]) {
```

Comment:
nit: may be easier to read with a function "wrapping" the `objMD.location && Array.isArray(objMD.location) && objMD.location[0]` expression behind a meaningful name (like `isRemoteObject` or whatever this condition means)

Reply:
Small addition to that, undefined is not an array so the third check handles the second one already.